### PR TITLE
libsyntax: small refactorings

### PR DESCRIPTION
**Pull request by topecongiro**
This PR consists of small refactorings to libsyntax.

### CODE REVIEW

File: src/libsyntax/parse/parser.rs [hunk 1]
```diff
@@ -1145,9 +1142,7 @@ impl<'a> Parser<'a> {
     {
         self.expect(bra)?;
         let result = self.parse_seq_to_before_end(ket, sep, f)?;
-        if self.token == *ket {
-            self.bump();
-        }
+        self.eat(ket);
```
Comment (petrochenkov) [inline src/libsyntax/parse/parser.rs]: This should change diagnostics by adding `ket` to the expected set, but it looks like a desirable change in this case.

File: src/libsyntax/parse/token.rs [hunk 2]
```diff
@@ -365,6 +365,10 @@ impl Token {
         self == &Lt || self == &BinOp(Shl)
     }
 
+    crate fn is_args_start(&self) -> bool {
+        self.is_qpath_start() || self == &OpenDelim(Paren)
```
Comment (petrochenkov) [inline src/libsyntax/parse/token.rs]: Well, the tokens technically coincide with the `is_qpath_start` set, but `<`/`<<` in `Type<T>`/`Type<<X as Y>::Assoc>` is not a qpath start at all.
I'd remove this commit.

File: src/libsyntax/parse/parser.rs [hunk 3]
```diff
@@ -1934,17 +1934,7 @@ impl<'a> Parser<'a> {
...
-            token::Interpolated(ref nt) => match nt.0 {
-                token::NtMeta(ref meta) => match meta.node {
-                    ast::MetaItemKind::Word => Some(meta.ident.clone()),
-                    _ => None,
-                },
-                _ => None,
-            },
-            _ => None,
-        };
-        if let Some(path) = meta_ident {
```
Comment (petrochenkov) [inline src/libsyntax/parse/parser.rs]: Outlining a small function that's used only once doesn't look like a good style to me.
I'd remove this commit.

**Comment by petrochenkov** [comment]
Thanks!
r=me after addressing the two comments above

**Comment by Aaronepower** [comment]
Triage; @topecongiro is there any update on this?

**Comment by pietroalbini** [comment]
Thank you for this PR @topecongiro! Unfortunately we haven't heard from you on this in a while, so I'm closing the PR to keep things tidy. Don't worry though, if you'll have time again in the future please **reopen** this PR, we'll be happy to review it again!

**Pull request closed without merging**